homeassistant/components/reolink/camera.py

```diff
@@ -30,25 +32,32 @@ async def async_setup_entry(
             streams.append("ext")
 
         for stream in streams:
-            cameras.append(ReolinkCamera(hass, config_entry, channel, stream))
+            cameras.append(ReolinkCamera(reolink_data, config_entry, channel, stream))
 
-    async_add_devices(cameras, update_before_add=True)
+    async_add_entities(cameras, update_before_add=True)
 
 
 class ReolinkCamera(ReolinkCoordinatorEntity, Camera):
     """An implementation of a Reolink IP camera."""
 
     _attr_supported_features: CameraEntityFeature = CameraEntityFeature.STREAM
 
-    def __init__(self, hass, config, channel, stream):
+    def __init__(
+        self,
+        reolink_data: ReolinkData,
+        config_entry: ConfigEntry,
+        channel: int,
+        stream: str,
+    ) -> None:
         """Initialize Reolink camera stream."""
-        ReolinkCoordinatorEntity.__init__(self, hass, config)
+        ReolinkCoordinatorEntity.__init__(self, reolink_data, config_entry)
         Camera.__init__(self)
 
         self._channel = channel
```

Comment:
I suggest we pass the channel to the `ReolinkCoordinatorEntity` instead and let it set the attribute properly as it already sets a default. Then we can also move the `device_info` property to an `_attr_device_info` attribute in the init method of `ReolinkCoordinatorEntity`.

Reply:
Done